### DISCUSSION TITLE
CMake config module for mitsuba3 ?

### DISCUSSION THREAD

**Discussion by nyue** [mitsuba-renderer/mitsuba3#148] [Feature requests]
Will there be cmake config module be created for Mitsuba in the future to enable building C++ extension/plugins for Mitsuba?

**Comment by dvicini** [comment]
Supporting external C++ plugins is on our more longer term TODO list. It shouldn't be too difficult, since each plugin already is its own shared library, but we haven't yet gotten around to support it. 

Part of the reason for that is that simply writing plugins in Python is very well supported at this point and leads to efficient execution when using one of the JIT modes. Eventually, having C++ plugins would still be useful though, it just hasn't been a very high priority for us.
